src/suricatta/dsl/pgsql.clj

```diff
@@ -45,7 +45,7 @@
 (extend-protocol proto/ISQLType
   java.sql.Array
   (-convert [self]
-    (into [] (.getArray self))))
+    (into [] (map proto/-convert (.getArray self)))))
```

Comment:
Seems like a perfect use for pass a transducer to `into` instead of create additional lazy seq from array.